[OSP] Support SR-IOV in UPI (#28315)
openshift/openshift-docs: installing/installing_openstack/installing-openstack-user-sr-iov-kuryr.adoc

// Not supported in 4.7.

[id="installing-openstack-user-sr-iov-kuryr"]
= Installing a cluster on OpenStack with Kuryr on your own SR-IOV infrastructure
include::modules/common-attributes.adoc[]
:context: installing-openstack-user-sr-iov-kuryr

toc::[]

> In {product-title} version {product-version}, you can install a cluster on

Author: ```suggestion
In {product-title} {product-version}, you can install a cluster on
```

{rh-openstack-first} that runs on user-provisioned infrastructure and uses SR-IOV networks to run compute machines.

> Using your own infrastructure allows you to integrate your cluster with existing infrastructure and modifications. The process requires more labor on your part than installer-provisioned installations, because you must create all {rh-openstack} resources, like Nova servers, Neutron ports, and security groups. However, Red Hat provides Ansible playbooks to help you in the deployment process.

Author: ```suggestion
Using your own infrastructure allows you to integrate your cluster with existing infrastructure and modifications. The process requires more labor on your part than installer-provisioned installations, because you must create all {rh-openstack} resources, such as Nova servers, Neutron ports, and security groups. However, Red Hat provides Ansible playbooks to help you in the deployment process.
```

== Prerequisites

* Review details about the
xref:../../architecture/architecture-installation.adoc#architecture-installation[{product-title} installation and update] processes.

> ** Verify that {product-title} {product-version} is compatible with your {rh-openstack} version in the _Available platforms_ section. You can also compare platform support across different versions by viewing the link:https://access.redhat.com/articles/4679401[{product-title} on {rh-openstack} support matrix].

Author: I believe it was moved and/or renamed--it is in our docs, though. Will need to find and replace.

* Verify that your network configuration does not rely on a provider network. Provider networks are not supported.

> * Have an {rh-openstack} account where you want to install {product-title}.

Author: ```suggestion
* Have a {rh-openstack} account where you want to install {product-title}.
```

> * On the machine from which you run the installation program, have:

Author: I'll workshop the second bullet. 

The assembly prereqs for ShiftStack are a bit clunky, but they're the result of a lot of painstaking negotiations. The account bullet being standalone is one of those things.

> * On the machine from which you run the installation program, have:

Author: ```suggestion
* On the machine where you run the installation program, have:
```
